packages/pwa-kit-dev/src/configs/webpack/config.js

```diff
@@ -50,6 +50,69 @@ export const EXT_OVERRIDES_DIR =
 export const EXT_OVERRIDES_DIR_NO_SLASH = EXT_OVERRIDES_DIR?.replace(/^\//, '')
 export const EXT_EXTENDS = pkg?.ccExtensibility?.extends
 export const EXT_EXTENDABLE = pkg?.ccExtensibility?.extendable
+// in a Template Extensibility project, these can include BOTH the node_modules/*
+// AND the node_modules/<extends>/node_modules/* npm deps, so we route here to
+// prevent duplication in the final bundle. We want a better way of doing this,
+// but for now this is the best option we've found.
+
+// TODO: can these be handled in package.json as peerDependencies?
+// https://salesforce-internal.slack.com/archives/C0DKK1FJS/p1672939909212589
+
+export const OVERRIDABLE_DEPS = [
```

Comment:
Many of these deps are from `template-retail-react-app`, I wonder if the code should be moved to the `template-retail-react-app` (instead of being implemented in `pwa-kit-dev` package) and add a mechanism for templates to declare the deps they need.

Reply:
A dumb idea, what if the `pwa-kit-dev` webpack config reads the dependencies required from the `extends` templates and automatically add alias for all of those deps.

Reply:
I worry about the risk of transitive dependencies in that scenario, but sure, I'm fine with that if it works.

Reply:
are there any transitive packages listed in this array?

Reply:
If someone adds a dep to `template-retail-react-app`, does he/she also need to add that dep to `OVERRIDABLE_DEPS`?

Reply:
Duplicates only happen from the monorepo, this is no longer relevant, as customer projects won't have this issue